KOGITO-1216 Bump services to Quarkus 1.3.0.CR2

tiagodolphine opened the pull request:
Unifying quarkus version , jackson on kogito-runtimes. In this way all kogito will use this version.
Adding the bom to specify netty version on all dependencies, change was made on kogito-quarkus-bom replacing the explicit dependencies that were added to fix the issue.

Related https://github.com/kiegroup/kogito-apps/pull/91

@evacchi 
@mariofusco 
@cristianonicolai

evacchi commented on pom.xml:
```diff
@@ -116,8 +117,8 @@
-    <version.org.infinispan>10.0.0.Final</version.org.infinispan>
-    <version.org.infinispan.image>10.0.0.Final-1</version.org.infinispan.image>
+    <version.org.infinispan>10.1.3.Final</version.org.infinispan>
+    <version.org.infinispan.image>10.1.3.Final</version.org.infinispan.image>
```
is this the version they are using in Quarkus?

tiagodolphine replied:
On quarkus it is set `10.1.2.Final` I'll change to keep the same.

https://github.com/quarkusio/quarkus/blob/1ec70f57f8710439ce73e0d9e49afe8f2aa8c183/bom/runtime/pom.xml#L122

evacchi commented on pom.xml:
```diff
@@ -107,7 +107,8 @@
     <version.io.prometheus>0.5.0</version.io.prometheus>
-    <version.io.quarkus>1.1.0.Final</version.io.quarkus>
+    <version.io.quarkus>1.3.0.Alpha2</version.io.quarkus>
+    <netty.version>4.1.46.Final</netty.version>
```
shall we use .45 ? this is the version they are using in Quarkus 1.3.0.CR1 AFAIK

tiagodolphine replied:
Yeah on quarkus it is set the .45, I'll keep the same but in fact, I'm not sure if this is mandatory since it was a bug previously so... keep the same version doesn't guarantee that nothing will be broken.

https://github.com/quarkusio/quarkus/blob/1ec70f57f8710439ce73e0d9e49afe8f2aa8c183/bom/runtime/pom.xml#L125

ricardozanini commented on addons/persistence/infinispan-quarkus-health-addon/src/test/resources/application.properties:
```diff
@@ -20,4 +20,6 @@ quarkus.infinispan-client.use-auth=true
\ No newline at end of file
+quarkus.infinispan-client.auth-server-name=infinispan
+quarkus.infinispan-client.client-intelligence=BASIC
+quarkus.infinispan-client.sasl-mechanism=DIGEST-MD5
```
Just a side note, last time I checked (some months ago), this SASL Mechanism wasn't working with the deployed Infinispan image using the theirs operator. Have you got the chance to test it? That's why our operator set it to `PLAIN`. If it's working on your side using their images, I'll give this another shot.

To test it, just deploy their Operator, create a new Infinispan instance and try to connect your local application to it using the credentials from the generated secret.

It it's not working, I'd recommend using `PLAIN` instead.

tiagodolphine replied:
@ricardozanini this configuration is used for the `test` scope, for the default scope we have a different application.properties. And if we can use the Digest on our deploy env I think it is better, the operator can override the parameter and if necessary we can change it to the default be the same.

ricardozanini replied:
Oh, I missed the `test` path in the file (my bad). I understand that `DIGEST` is way better, that's why I asked you to run through some tests if possible using their image, since wasn't working last time I checked.

tiagodolphine commented:
@evacchi thanks for the review. On the `kogito-quarkus-extension` module I already removed the properties, and I found another point setting it on the `archetypes` module, and I removed it as well to use the same as the runtimes parent, is it ok or you see any problem?

tiagodolphine commented:
@evacchi @cristianonicolai anything else to add here? 
Thanks.

tiagodolphine commented:
The **1.3.0.CR1** is available, so I'll update this PR to it, is ok for you guys?

evacchi commented:
good catch for the archetype. Cool for me to go with 1.3.0.CR1. All good

evacchi commented on kogito-quarkus-extension/bom/pom.xml:
just remembered, if you upgrade to CR1 you can remove this, and close KOGITO-1315, KOGITO-1369

tiagodolphine replied:
+1

evacchi commented:
CR2 is out!

tiagodolphine commented:
@evacchi ouch, let's bump again and run the tests.. :)
Thanks for warning me.

tiagodolphine commented:
@evacchi @cristianonicolai updated quarkus to **1.3.0.CR2**

cristianonicolai commented:
@evacchi looks good to you ?

@tiagodolphine should we get this merged and create a follow-up JIRA to update once 1.3.0 final is released? I would way for that to release 0.8.1

evacchi commented:
@cristianonicolai I agree, let's get this merged 
@tiagodolphine there's a conflict

tiagodolphine commented:
@evacchi I'll fix the conflict and let you know when it is done.

tiagodolphine commented:
@evacchi all green

ricardozanini commented:
Please merge this :)

Will make my life easier to test this: https://github.com/quarkusio/quarkus/issues/7791
